src/decimal.rs

```diff
@@ -504,14 +513,7 @@ impl Decimal {
     /// * `mid` - The middle 32 bits of a 96-bit integer.
     /// * `hi` - The high 32 bits of a 96-bit integer.
     /// * `negative` - `true` to indicate a negative number.
-    /// * `scale` - A power of 10 ranging from 0 to 28.
-    ///
-    /// # Caution: Undefined behavior
-    ///
-    /// While a scale greater than 28 can be passed in, it will be automatically capped by this
-    /// function at the maximum precision. The library opts towards this functionality as opposed
-    /// to a panic to ensure that the function can be treated as constant. This may lead to
-    /// undefined behavior in downstream applications and should be treated with caution.
```

Comment:
Whilst reviewing docs for changes I also spotted that we can remove this undefined behaviour in `from_parts` and `from_parts_raw`. Given that we describe it as undefined behaviour previously, and a `panic` is a valid result (IMO) from undefined behaviour, it _could_ be argued this is not a breaking change. 

It's also worth noting that the documentation on what the function would do in this case (automatically capped by this function at the maximum precision) is not what it did anyway (it instead found the remainder past the end of the scale).

Note that the random generation functions needed fixing as they relied on this scale, and there was one additional test failure: `issue_384_neg_overflow_during_subtract_carry` hits the new `assert` for `from_parts`. I just calculated what the value would have been with the old behaviour (`3873892070 % 29 == 5`) to resolve this.